src/Logout.tsx

```diff
@@ -46,8 +54,9 @@ const Logout: FC<Props> = ({history}) => {
         history.push(`/signin`)
       }
     }
-    dispatch(reset())
-    dispatch({type: 'USER_LOGGED_OUT'})
+    if (!isFlagEnabled('authSessionCookieOn')) {
+      handleReset()
```

Comment:
Why are we checking before running `handleReset()`?  It looks like it runs here if the `authSessionCookieOn` is false but runs if that is true on either 39 (if cloud is true) or 46 (if cloud is false).  So doesn't `handleReset()` just always get called?

Reply:
it does always get called. 

> This PR closes the issue where users logging out using the shared session cookie were undergoing the original logout loop. This issue presented itself because the feature flag and local storage were being reset prior to hitting the condition of checking the feature flag. As such, we are now checking to see if the flag is enabled prior to resetting it

Reply:
So is it important what other things run first?  I'm mostly just confused why we aren't just calling `handleReset()` at this point.

Reply:
Because calling reset before calling the signout was resetting our feature flags and we weren't hitting the feature flagged condition inside the logout function